owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/OCSettingsPasscode.kt

```diff
@@ -0,0 +1,226 @@
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class OCSettingsPasscode {
+
+    @Rule
+    @JvmField
+    val activityRule = ActivityTestRule(PassCodeActivity::class.java, true, false)
+
+    val intent = Intent()
+    val errorMessage = "PassCode Activity error"
+    private val KEY_PASSCODE = "KEY_PASSCODE"
+    private val context = InstrumentationRegistry.getInstrumentation().targetContext
+
+    @After
+    fun tearDown() {
+        //Clean preferences
+        PreferenceManager.getDefaultSharedPreferences(context).edit().clear().commit()
+    }
+
+
+    @Test
+    fun passcodeView(){
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        onView(withId(R.id.header)).check(matches(isDisplayed()))
+        onView(withId(R.id.explanation)).check(matches(isDisplayed()))
+        onView(withText(R.string.pass_code_configure_your_pass_code)).check(matches(isDisplayed()))
+        onView(withText(R.string.pass_code_configure_your_pass_code_explanation)).check(matches(isDisplayed()))
+        onView(withId(R.id.txt0)).check(matches(isDisplayed()))
+        onView(withId(R.id.txt1)).check(matches(isDisplayed()))
+        onView(withId(R.id.txt2)).check(matches(isDisplayed()))
+        onView(withId(R.id.txt3)).check(matches(isDisplayed()))
+        onView(withId(R.id.cancel)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun firstTry(){
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //First typing
+        typePasscode(arrayOf('1','1','1','1'))
+
+        onView(withText(R.string.pass_code_reenter_your_pass_code)).check(matches(isDisplayed()))
+        onView(withText(R.string.pass_code_configure_your_pass_code)).check(doesNotExist())
+    }
+
+    @Test
+    fun secondTryCorrect(){
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //First typing
+        typePasscode(arrayOf('1','1','1','1'))
+        //Second typing
+        typePasscode(arrayOf('1','1','1','1'))
+
+        //Checking that the setResult returns the typed passcode
+        assertThat(activityRule.getActivityResult(), hasResultCode(Activity.RESULT_OK))
+        assertThat(activityRule.getActivityResult(), hasResultData(hasExtra(KEY_PASSCODE, "1111")))
+
+        assertTrue(errorMessage, activityRule.activity.isFinishing)
+    }
+
+    @Test
+    fun secondTryIncorrect(){
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //First typing
+        typePasscode(arrayOf('1','1','1','1'))
+        //Second typing
+        typePasscode(arrayOf('1','1','1','2'))
```

Comment:
Save `arrayOf('1','1','1','2')` in a constant at the top of the class

Reply:
Done 👍